cmd/restic/cmd_restore_integration_test.go

```diff
@@ -144,19 +223,38 @@ func TestRestoreFilter(t *testing.T) {
 		rtest.OK(t, testFileSize(filepath.Join(env.base, "restore0", "testdata", testFile.name), int64(testFile.size)))
 	}
 
-	for i, pat := range []string{"*.c", "*.exe", "*", "*file3*"} {
-		base := filepath.Join(env.base, fmt.Sprintf("restore%d", i+1))
-		testRunRestoreExcludes(t, env.gopts, base, snapshotID, []string{pat})
-		for _, testFile := range testfiles {
-			err := testFileSize(filepath.Join(base, "testdata", testFile.name), int64(testFile.size))
-			if ok, _ := filter.Match(pat, filepath.Base(testFile.name)); !ok {
-				rtest.OK(t, err)
-			} else {
-				rtest.Assert(t, os.IsNotExist(err),
-					"expected %v to not exist in restore step %v, but it exists, err %v", testFile.name, i+1, err)
+	excludePatterns := []string{"*.c", "*.exe", "*", "*file3*"}
+
+	testRestoreFileExclusions := func(t *testing.T, env *testEnvironment, excludePatterns []string) {
+		for i, pat := range excludePatterns {
+			base := filepath.Join(env.base, fmt.Sprintf("restore%d", i+1))
+			testRunRestoreExcludes(t, env.gopts, base, snapshotID, []string{pat})
+			for _, testFile := range testfiles {
+				err := testFileSize(filepath.Join(base, "testdata", testFile.name), int64(testFile.size))
+				if ok, _ := filter.Match(pat, filepath.Base(testFile.name)); !ok {
+					rtest.OK(t, err)
+				} else {
+					rtest.Assert(t, os.IsNotExist(err),
+						"expected %v to not exist in restore step %v, but it exists, err %v", testFile.name, i+1, err)
+				}
 			}
 		}
 	}
+
+	testRestoreFileExclusions(t, env, excludePatterns)
+
+	// Create an include file with some patterns
+	patternsFile := env.base + "/patternsFile"
+	fileErr := os.WriteFile(patternsFile, []byte(strings.Join(excludePatterns, "\n")), 0644)
+	if fileErr != nil {
+		t.Fatalf("Could not write include file: %v", fileErr)
+	}
+
+	restoredir := filepath.Join(env.base, "restore-exclude-from-file")
+
+	testRunRestoreExcludesFromFile(t, env.gopts, restoredir, snapshotID, patternsFile)
+
+	testRestoreFileExclusions(t, env, excludePatterns)
```

Comment:
This call performs multiple restores of its own, but does not use the exclude file.

Reply:
Fixed, I made the logic simpler and similar to the inclusion tests.